### PR TITLE
Store static type in array/dictionary values

### PR DESCRIPTION
**Pull request by SupunS**
Work towards #870

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: runtime/interpreter/value.go [hunk 1]
```diff
@@ -656,8 +658,6 @@ func NewDeferredArrayValue(
 	version uint16,
 ) *ArrayValue {
 	return &ArrayValue{
-		// TODO: type
-		Type:            nil,
```
Comment (turbolent) [inline runtime/interpreter/value.go:660]: I added this TODO so we wouldn't forget to add it here. 
Or will the type never be available here and later lazily decoded?

File: runtime/interpreter/value_test.go [hunk 2]
```diff
@@ -1627,22 +1637,22 @@ func TestArrayValue_Equal(t *testing.T) {
 
 	t.Parallel()
 
+	uint8ArrayStaticType := VariableSizedStaticType{
```
Comment (turbolent) [inline runtime/interpreter/value_test.go:1640]: 👌

File: runtime/tests/interpreter/interpreter_test.go [hunk 3]
```diff
@@ -8241,29 +8241,25 @@ func TestInterpretInternalAssignment(t *testing.T) {
 	value, err := inter.Invoke("test")
 	require.NoError(t, err)
 
+	stringIntDictionaryStaticType := interpreter.DictionaryStaticType{
```
Comment (turbolent) [inline runtime/tests/interpreter/interpreter_test.go:8244]: 👌

**Review by turbolent — APPROVED** [review]

**Comment by turbolent** [comment]
I'm fixing the conflicts and will push them

File: runtime/interpreter/value.go (hunk 1 above)
Comment (SupunS) [reply runtime/interpreter/value.go:677]: Yeah, it will be added later, when the value is lazily decoded. I added it in #1035, so removed it from here.

**Pull request merged**